assessments-service/src/main/resources/db/migration/common/V1.43__update_indexes.sql

```diff
@@ -0,0 +1,7 @@
+ALTER TABLE `Assessment` ADD INDEX `assessment_soft_deleted_date` (`softDeletedDate`);
+ALTER TABLE `AssessmentOutcome` ADD INDEX `assessment_outcome_outcome` (`outcome`);
+ALTER TABLE `Assessment` ADD INDEX `assessment_duplicates`
+(`traineeId`, `programmeMembershipId`, `reviewDate`, `softDeletedDate`);
+ALTER TABLE `Assessment` ADD INDEX `assessment_search`
+(`type`, `gmcNumber`, `gdcNumber`, `publicHealthNumber`);
```

Comment:
I've got some opinion that don't use indexes on columns with only few different values, see here (the 1st anser): https://stackoverflow.com/questions/13441334/when-to-add-an-index-on-a-sql-table-field-mysql/13441522. 
So probably we don't need add index on `type`?

Reply:
Ah, that's a good tip I hadn't considered, thanks! 

I can see 38 distinct values for type, though ARCP and RITA cover about 90% of them... as such, I don't know if type falls into the category of 'only a few values' or not, to be honest?

Another thing I didn't think of when defining the composite index is the ordering of the fields. I think its ok for the assessments_duplicates index (and in fact would mean we could drop the index on traineeId alone if we wanted to, since the composite index will be used), but for the assessments_search index I think gmcNumber is the field that discriminates the best among the data, so afaik it should be first... I'll resubmit with a revised ordering.